coderd/rbac/authz.go

```diff
@@ -106,7 +122,51 @@ func NewAuthorizer() *RegoAuthorizer {
 			panic(xerrors.Errorf("compile rego: %w", err))
 		}
 	})
-	return &RegoAuthorizer{query: query}
+
+	// Register metrics to prometheus.
+	// These bucket values are based on the average time it takes to run authz
+	// being around 1ms. Anything under ~2ms is OK and does not need to be
+	// analyzed any further.
+	buckets := []float64{
+		0.0005, // 0.5ms
+		0.001,  // 1ms
+		0.002,  // 2ms
+		0.003,
+		0.005,
+		0.01, // 10ms
+		0.02,
+		0.035, // 35ms
+		0.05,
+		0.075,
+		0.1,  // 100ms
+		0.25, // 250ms
+		0.75, // 750ms
+		1,    // 1s
+	}
+
+	factory := promauto.With(registry)
+	authorizeHistogram := factory.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "coderd",
+		Subsystem: "authz",
+		Name:      "authorize_duration_s",
```

Comment:
Could we use milliseconds here? I think it makes a bit more sense given the scale caps out at 1s.

Reply:
Prometheus docs suggest using base units, and for time that is seconds: https://prometheus.io/docs/practices/naming/#base-units

You can change it to `ms` in whatever graphing tool you use.

Reply:
Nit: if we're following the Prometheus docs, we should call it `authorize_duration_seconds`

Reply:
If we were doing this completely from scratch I'd agree, but we already have a number of metrics measured in milliseconds e.g. `api_request_duration_ms`. I would argue for consistency here instead of 'correctness'.